src/render_tree/text.rs

```diff
@@ -1,20 +1,21 @@
-static DEFAULT_FONT: &str = "Times New Roman";
+use std::{ffi::CString, os::raw::c_char};
 
-#[derive(Debug, PartialEq)]
+#[derive(Debug, PartialEq, Clone, Copy)]
+#[repr(C)]
 pub struct TextNode {
     /// Body of the text node that will be drawn
-    pub value: String,
-    pub font: String,
-    pub font_size: f64,
+    pub value: *mut c_char,
+    pub font: *mut c_char,
```

Comment:
Rust strings are not compatible with C, and using `CString` is too inconvenient since we need to append text, so storing raw pointers here is the most convenient

Reply:
I don't really like having a raw ptr somewhere in the engine. Are you sure, you can't do something like a `cstring_append` function?

Reply:
I did not finding any appending functionality with CString. The only alternative I can think of at the moment is storing them as strings in the engine, but in the bindings converting them to a different struct that will convert String to the raw pointer and transfer ownership to the C API. That might be doable

Reply:
Ah, just saw your other message below, I guess we had similar thinking

Reply:
Looking through the `String` docs, looks like we can convert it to a box pointer and from there convert it to a raw ptr

`Box::into_raw(my_string.into_boxed_str())`

So maybe for the `CTextNode` we can implement a `From` trait that takes a regular `TextNode` containing the strings and perform the conversions to send back to C